architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,580 @@
+
+The following statements are all accurate:
+
+- The `System.Resources/resourceProviders` type is the parent of `System.Resources/resourceProviders/resourceTypes`. 
+- The `System.Resources/resourceProviders/resourceTypes` type is the child of `System.Resources/resourceProviders`.
+- The `System.Resources/resourceProviders/resourceTypes` type is nested within `System.Resources/resourceProviders`.
+- The `System.Resources/resourceProviders` type is a top-level resource type.
+
+Any resource type with two segments (separated by `/`) is a top-level resource type. Any resource type with 3 or more segments (separate by `/`) is a nested resource type.
+
+The resource ids (and URLs) are written like the following:
+
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders` - List all resource providers.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example` - Get the Contoso.Example resource provider.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example/resourceTypes` - List all resource types inside the `Contoso.Example` resource provider/namespace.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example/resourceTypes/contosoBuses` - Gets the `contosoBuses` resource type inside the `Contoso.Example` resource provider/namespace.
+
+Nesting a resource type also nests the lifecyle of the resource. Using our examples above, the `/planes/radius/{plane}/providers/System.Resources/resourceProviders/resourceTypes/Contoso.Example/contosoBuses` type cannot exist unless `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example` already exists.
+
+There are a few simple rules for this:
+
+- For operations on parent resources...
+  - A `DELETE` operation on a parent resource deletes all children.
+
+- For operations on child resources...
+  - A *CRUL* operation (anything but delete) on a child resource must return a `404` when the parent does not exist.
+  - A `DELETE` operation on a child resource never returns a `404`.
+
+### API design (if applicable)
+
+We should consider the set of concepts that need to be included in the resource provider and resource type APIs. For this design doc we'll consider the minimal set necessary to route to one of our existing resource providers. As we build more features for user-defined types we'll add more functionality. We don't want to design functionality in the API separately from its usage -- an incremental approach is better.
+
+The set of concepts needed right now looks like this:
+
+Resource providers:
+
+- Locations: the set of logical locations where the resource provider is supported. Each location has:
+  - An optional address/URL of the resource provider. This is optional because it won't be needed for declarative user-defined types.
+
+
+Resource types:
+
+- Locations: the set of logical locations where the resource type is supported.
```

Comment:
I can understand this if we are using cloud resources but if we are not, how do we use the locations for UCP and/or resource providers and/or resource types?

Reply:
Discussed in the meeting. My action item is to make a more concrete description of what location **means**.

Short answer for now: location determines which instance of the control-plane is responsible. It does not constrain what users can do. 

eg: a `westus3` instance of Radius can manage Azure resources in any Azure location.